translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+    if info['metadata'].get('datafeed_frequency_update', False):
+        wzd['road_event_feed_info']['update_frequency'] = info['metadata'][
+            'datafeed_frequency_update']  # Verify data type
+    wzd['road_event_feed_info']['version'] = '3.0'
+
+    data_source = {}
+    data_source['data_source_id'] = str(uuid.uuid4())
+    data_source['feed_info_id'] = info['feed_info_id']
+    data_source['organization_name'] = info['metadata']['issuing_organization']
+    data_source['contact_name'] = info['metadata']['contact_name']
+    data_source['contact_email'] = info['metadata']['contact_email']
+    if info['metadata'].get('datafeed_frequency_update', False):
+        data_source['update_frequency'] = info['metadata']['datafeed_frequency_update']
+    data_source['update_date'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+    data_source['location_method'] = info['metadata']['wz_location_method']
+    data_source['lrs_type'] = info['metadata']['lrs_type']
+    wzd['road_event_feed_info']['data_sources'] = [data_source]
+
+    wzd['type'] = 'FeatureCollection'
+    sub_identifier = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in
+                             range(6))  # Create random 6 character digit/letter string
+    road_event_id = str(uuid.uuid4())
+    ids = {}
+    ids['sub_identifier'] = sub_identifier
+    ids['road_event_id'] = road_event_id
+
+    wzd['features'] = []
+
+    if not messages.get('alert:Alerts', {}).get('alert:Alert'):
+        return None
+
+    for alert in messages['alert:Alerts']['alert:Alert']:
+        # Parse alert to WZDx Feature    
+        feature = parse_alert(alert, callback_function=unsupported_message_callback)
+        if feature:
+            wzd['features'].append(feature)
+    if not wzd['features']:
+        return None
+    wzd = add_ids(wzd, True)
+    return wzd
+
+
+
+def validate_info(info):
+
+    if ((not info) or (type(info) != dict and type(info) != OrderedDict)):
+        logging.warning('invalid type')
+        return False
+    
+    #### Consider whether this id needs to be hardcoded or generated
+    feed_info_id = str(info.get('feed_info_id', ''))
+    check_feed_info_id = re.match('[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}', feed_info_id)
+    #### This information is required, might want to hardcode
+    metadata=info.get('metadata', {})
+    wz_location_method = metadata.get('wz_location_method')
+    lrs_type = metadata.get('lrs_type')
+    contact_name = metadata.get('contact_name')
+    contact_email = metadata.get('contact_email') 
+    issuing_organization=metadata.get('issuing_organization')
+    required_fields = [ check_feed_info_id, metadata, wz_location_method, lrs_type, contact_name, contact_email, issuing_organization]
+    for field in required_fields:
+        if not field:
+            return False
+            logging.warning( 'Not all required fields are present') 
```

Comment:
This logging statement is never hit, we return before it is called.

Reply:
i moved the logging statement to its proper position